ssm/hmm/transitions.py

```diff
@@ -162,7 +162,7 @@ def __init__(self,
         # If the transition matrix is given, recompute it given stay_probability.
         if transition_distribution is None:
             assert (stay_probability >= 0) and (stay_probability <= 1)
-            self.stay_probability = stay_probability
+            self.stay_probability = np.float32(stay_probability)  # prevent weak typing
```

Comment:
Kind of an interesting detail here. EM was double jitting because of the weak typing for `stay_probability` (specified as a Python float, not a JAX float32). By explicitly specifying here we can avoid re-jit.